hazelcast/src/main/java/com/hazelcast/instance/impl/DefaultNodeExtension.java

```diff
@@ -371,17 +372,31 @@ protected PartitioningStrategy getPartitioningStrategy(ClassLoader configClassLo
             return createMapService();
         } else if (JetServiceBackend.class.isAssignableFrom(clazz)) {
             return (T) new JetServiceBackend(node);
+        } else if (ClusterWideConfigurationService.class.isAssignableFrom(clazz)) {
+            return createConfigurationService(params[0]);
         }
 
         throw new IllegalArgumentException("Unknown service class: " + clazz);
     }
 
+    @SuppressWarnings("unchecked")
     private <T> T createMapService() {
         ConstructorFunction<NodeEngine, MapService> constructor = getDefaultMapServiceConstructor();
         NodeEngineImpl nodeEngine = node.getNodeEngine();
         return (T) constructor.createNew(nodeEngine);
     }
 
+    @SuppressWarnings("unchecked")
+    private <T> T createConfigurationService(Object nodeEngine) {
+        if (!(nodeEngine instanceof NodeEngine)) {
+            throw new IllegalArgumentException(
+                    "While creating ConfigurationService expected NodeEngine as a parameter, but found: "
+                            + nodeEngine.getClass().getName()
```

Comment:
Can't we have a compile-time check here instead of runtime check? So to accept `NodeEngine` typed params instead of `Object`.

Reply:
I believe we can do that, but in the future if somebody else also need to pass another parameter like I do (for example Node object or some other service etc.), passing Object is better I believe.

Reply:
But if that's needed, we can open the accepted types. The consumer of the parameter is fixed to a certain type, isn't it? So if we need to change the type, both the caller and the called site has to be modified. Am I overlooking something?

Reply:
I see this is called from `createService` where it makes sense to accept any type as parameter, so we need a runtime type check anyways. My question ignored that part, it's good as is. Thanks for pointing this out :+1: